src/pqueue.js

```diff
@@ -0,0 +1,102 @@
+// 'use strict'
+//
+// class Node {
+//   constructor ( value, priority ){
+//     this.arr = []
+//     this.value = value
+//     this.priority = priority
+//   }
+// }
+
+export default class PriorityQueue {
+  constructor(value) {
+    this.arr = value || []
+    this.index = 0
+  }
+
+  enqueue(data, priority) {
+    return this.arr[this.index++] = {data, priority}
+  }
+
+  front() {
+    let current,
+        highestPriority,
+        prev
+
+    if (this.index === 0) {
+      return null
+    }
+    for (let i = 1; i < this.index; i++) {
+      this.current = this.arr[i]
```

Comment:
You should not be using an instance variable to store this data - `current` is not an attribute of the `PriorityQueue`, it's a local variable used to maintain state of the iteration in the context of this loop.
```js
let current = this.arr[ i ]
```
The same is true for successive uses of `prev` and `highestPriority`.